pyomo/core/expr/ndarray.py

```diff
@@ -0,0 +1,39 @@
+#  ___________________________________________________________________________
```

Comment:
This was completely moved and yet seemed to affect no tests? That concerns me.

Reply:
It did affect tests: see changes in [pyomo/core/tests/unit/test_numpy_expr.py](https://github.com/Pyomo/pyomo/pull/3077/files#diff-cb72e9fbedeaeb4caef81b79ce8b12e73d3a8f4033266290cc2b83cebca2ca7c)